fuel/datasets/cifar10.py

```diff
@@ -33,23 +34,12 @@ class CIFAR10(H5PYDataset):
     filename = 'cifar10.hdf5'
 
     def __init__(self, which_set, **kwargs):
-        if which_set not in ('train', 'test'):
-            raise ValueError("CIFAR10 only has a train and test set")
-
         kwargs.setdefault('load_in_memory', True)
-        self.which_set = which_set
-
-        super(CIFAR10, self).__init__(self.data_path, which_set,
-                                      **kwargs)
+        super(CIFAR10, self).__init__(self.data_path, which_set, **kwargs)
+        self.default_transformers += (
```

Comment:
Actually, I'm wondering if we even need things like the `flatten` constructor argument in `H5PYDataset` anymore. It's nothing more than a convenience flag for applying a transformer.

Is it reasonable to expect that users that want to flatten topological data do so explicitly on their own when constructing the data stream?

If so, then we could restrict the use default transformers to cases where the data is stored in a certain way but it makes more sense to use it under another representation, which would work well as a class attribute.